pkg/k8s/watchers/pod.go

```diff
@@ -737,15 +737,21 @@ func (k *K8sWatcher) deleteHostPortMapping(pod *slim_corev1.Pod, podIPs []string
 }
 
 func (k *K8sWatcher) updatePodHostData(oldPod, newPod *slim_corev1.Pod, oldPodIPs, newPodIPs k8sTypes.IPSlice) error {
+	logger := log.WithFields(logrus.Fields{
+		logfields.K8sPodName:   newPod.ObjectMeta.Name,
+		logfields.K8sNamespace: newPod.ObjectMeta.Namespace,
+	})
+
 	if newPod.Spec.HostNetwork {
-		logger := log.WithFields(logrus.Fields{
-			logfields.K8sPodName:   newPod.ObjectMeta.Name,
-			logfields.K8sNamespace: newPod.ObjectMeta.Namespace,
-		})
 		logger.Debug("Pod is using host networking")
 		return nil
 	}
 
+	if option.Config.EnableHighScaleIPcache && nodeTypes.GetName() != newPod.Spec.NodeName {
+		logger.Debug("Pod is not local; skipping ipcache upsert")
```

Comment:
nit: Pod watcher should only be listening for local pods anyway right? :thinking:

Reply:
If you look at the callers of `(k *K8sWatcher) createPodController`, there are 2.

1. Only listens for pods on the local node
2. Listens for all pods on all nodes

(2) occurs when `disable-endpoint-crd: true` and presumably with kvstore mode. So I think we need this code (conceptually) at least to cover for (2).

Reply:
On the other hand, we can say that kvstore mode (and disable-endpoint-crd: true) as not supported configuration when high-scale ipcache mode is enabled, and completely drop this change here, so that only case (1) is relevant.